创建vpc报错但是成功创建，用最新的插件和1.59.0都不行

[root@amj terraform-ws]# terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # huaweicloud_vpc.example will be created
  + resource "huaweicloud_vpc" "example" {
      + cidr                  = "192.168.0.0/16"
      + enterprise_project_id = (known after apply)
      + id                    = (known after apply)
      + name                  = "terraform_vpc"
      + region                = (known after apply)
      + routes                = (known after apply)
      + status                = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

huaweicloud_vpc.example: Creating...
huaweicloud_vpc.example: Still creating... [10s elapsed]
huaweicloud_vpc.example: Still creating... [20s elapsed]
huaweicloud_vpc.example: Still creating... [30s elapsed]
╷
│ Error: Plugin did not respond
│ 
│   with huaweicloud_vpc.example,
│   on main.tf line 2, in resource "huaweicloud_vpc" "example":
│    2: resource "huaweicloud_vpc" "example" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-huaweicloud_v1.59.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xe68554]

goroutine 30 [running]:
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/vpc.resourceVirtualPrivateCloudRead({0x40002536c0?, 0x1e530d3?}, 0x400118cb80, {0x1e35dc0?, 0x400121d200})
        github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/vpc/resource_huaweicloud_vpc.go:228 +0x9b4
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/vpc.resourceVirtualPrivateCloudCreate({0x22307b0, 0x4000266000}, 0x400118cb80, {0x1e35dc0?, 0x400121d200?})
        github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/vpc/resource_huaweicloud_vpc.go:165 +0xc30
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x40008e8700, {0x2230708, 0x4000d23a40}, 0xd?, {0x1e35dc0, 0x400121d200})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:707 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x40008e8700, {0x2230708, 0x4000d23a40}, 0x4000b7c820, 0x400118ca00, {0x1e35dc0, 0x400121d200})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:837 +0x850
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x400000c180, {0x2230708?, 0x4000d23950?}, 0x4000d4eb40)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1021 +0xb08
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x40004bdae0, {0x2230708?, 0x4000d23140?}, 0x4000a357a0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:818 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1d8bb20?, 0x40004bdae0}, {0x2230708, 0x4000d23140}, 0x4000a35730, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x40002b01e0, {0x2235658, 0x4000b72d00}, 0x4000efe5a0, 0x4000b76390, 0x343d8a0, 0x0)
        google.golang.org/grpc@v1.56.3/server.go:1335 +0xbe0
google.golang.org/grpc.(*Server).handleStream(0x40002b01e0, {0x2235658, 0x4000b72d00}, 0x4000efe5a0, 0x0)
        google.golang.org/grpc@v1.56.3/server.go:1712 +0x7cc
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.56.3/server.go:947 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 10
        google.golang.org/grpc@v1.56.3/server.go:958 +0x160

Error: The terraform-provider-huaweicloud_v1.59.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

[root@amj terraform-ws]# terraform destory
Terraform has no command named "destory". Did you mean "destroy"?

To see all of Terraform's top-level commands, run:
  terraform -help

--- a/huaweicloud/config/hc_config.go
+++ b/huaweicloud/config/hc_config.go
@@ -110,23 +110,23 @@ func buildHTTPConfig(c *Config) *hcconfig.HttpConfig {
 		AddResponseHandler(logResponseHandler)
 	httpConfig = httpConfig.WithHttpHandler(httpHandler)
 
-	if proxyURL := getProxyFromEnv(); proxyURL != "" {
-		if parsed, err := url.Parse(proxyURL); err == nil {
-			logp.Printf("[DEBUG] using https proxy: %s://%s", parsed.Scheme, parsed.Host)
+	if proxyURL, err := parseProxyFromEnv(); err == nil {
+		if proxyURL != nil {
+			logp.Printf("[DEBUG] using https proxy: %s://%s", proxyURL.Scheme, proxyURL.Host)
 
 			httpProxy := hcconfig.Proxy{
-				Schema:   parsed.Scheme,
-				Host:     parsed.Host,
-				Username: parsed.User.Username(),
+				Schema:   proxyURL.Scheme,
+				Host:     proxyURL.Host,
+				Username: proxyURL.User.Username(),
 			}
-			if pwd, ok := parsed.User.Password(); ok {
+			if pwd, ok := proxyURL.User.Password(); ok {
 				httpProxy.Password = pwd
 			}
 
 			httpConfig = httpConfig.WithProxy(&httpProxy)
-		} else {
-			logp.Printf("[WARN] parsing https proxy failed: %s", err)
 		}
+	} else {
+		logp.Printf("[WARN] parsing https proxy failed: %s", err)
 	}
 
 	return httpConfig
@@ -358,18 +358,37 @@ func implNewHcClient(c *Config, region, product string, isGlobal, isDerived bool
 	return builder.Build().PreInvoke(headers), nil
 }
 
-func getProxyFromEnv() string {
-	var url string
+func parseProxyFromEnv() (*url.URL, error) {
+	var proxy string
 
 	envNames := []string{"HTTPS_PROXY", "https_proxy"}
 	for _, n := range envNames {
 		if val := os.Getenv(n); val != "" {
-			url = val
+			proxy = val
 			break
 		}
 	}
 
-	return url
+	if proxy == "" {
+		return nil, nil
+	}
+
+	proxyURL, err := url.Parse(proxy)
+	if err != nil ||
+		(proxyURL.Scheme != "http" &&
+			proxyURL.Scheme != "https" &&
+			proxyURL.Scheme != "socks5") {
+		// proxy was bogus. Try prepending "http://" to it and
+		// see if that parses correctly. If not, we fall
+		// through and complain about the original one.
+		if proxyURL, err := url.Parse("http://" + proxy); err == nil {
+			return proxyURL, nil
+		}
+	}
+	if err != nil {
+		return nil, fmt.Errorf("invalid https proxy address %q: %v", proxy, err)
+	}
+	return proxyURL, nil
 }
 
 func logRequestHandler(request http.Request) {

